### ISSUE TITLE
How to reproduce image seen in fakesNNNNNN.png snapshot?

### ISSUE THREAD

**Issue by estan** [NVlabs/stylegan2-ada-pytorch#158]
I'd like to reproduce (generate anew) an image found in the fakesNNNNNN.png snapshot grid of my trained network, but I'm not sure how to figure out which random vectors each image in the grid corresponds to. Any help is much appreciated. I guess this is not possible out of the box using generate.py?

**Comment by YukiSakuma** [comment]
You could try this code on generating fakes I just copied it from training_loop.py, but I'm not able to reproduce the actual target fakes (using the default seed=0)

```
import datetime
import math
import os
import numpy as np
import PIL.Image
import torch
import dnnlib
import legacy
from torch_utils import misc


def save_image_grid(img, fname, drange, grid_size):
    lo, hi = drange
    img = np.asarray(img, dtype=np.float32)
    img = (img - lo) * (255 / (hi - lo))
    img = np.rint(img).clip(0, 255).astype(np.uint8)

    gw, gh = grid_size
    _N, C, H, W = img.shape
    img = img.reshape(gh, gw, C, H, W)
    img = img.transpose(0, 3, 1, 4, 2)
    img = img.reshape(gh * H, gw * W, C)

    assert C in [1, 3]
    if C == 1:
        PIL.Image.fromarray(img[:, :, 0], 'L').save(fname)
    if C == 3:
        PIL.Image.fromarray(img, 'RGB').save(fname)


def main():
    np.random.seed(seed * num_gpus + rank)
    torch.manual_seed(seed * num_gpus + rank)
    with dnnlib.util.open_url(network_pkl) as f:
        resume_data = legacy.load_network_pkl(f) 
        G_ema = resume_data['G_ema'].to(device)
        G = resume_data['G'].to(device)
    if rank == 0: #for some reason if this statement is commented fake output will be different
        z = torch.empty([batch_gpu, G.z_dim], device=device)
        c = torch.empty([batch_gpu, G.c_dim], device=device)
        img = misc.print_module_summary(G, [z, c])
    if rank == 0:
        num_images = 120
        grid_size = (15, 8)
        grid_z = torch.randn([num_images, G.z_dim], device=device).split(batch_gpu)
        labels = np.array([])
        grid_c = torch.from_numpy(labels).to(device).split(batch_gpu) * int(math.ceil(num_images / batch_gpu))
        images = torch.cat(
            [G_ema(z=z, c=c, noise_mode='const', truncation_psi=truncation_psi).cpu() for z, c in zip(grid_z, grid_c)]).numpy()
        save_image_grid(images, os.path.join(outdir, f'fakes_init-{seed}-psi-{truncation_psi}-{start_time}.jpg'), drange=[-1, 1],
                        grid_size=grid_size)


if __name__ == '__main__':
    network_pkl = 'model path'
    outdir = 'generate_fake_init'
    batch_gpu = 4
    num_gpus = 1
    rank = 0
    seed = 0
    truncation_psi = 1.0 
    device = torch.device('cuda', rank)
    os.makedirs(outdir, exist_ok=True)
    start_time = datetime.datetime.now().microsecond
    main()

```
